delfin/drivers/dell_emc/power_store/power_store.py

```diff
@@ -0,0 +1,167 @@
+from delfin.common import constants
+from delfin.drivers import driver
+from oslo_log import log
+
+from delfin.drivers.dell_emc.power_store import rest_handler, consts
+
+LOG = log.getLogger(__name__)
+
+
+class PowerStoreDriver(driver.StorageDriver):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rest_handler = rest_handler.RestHandler(**kwargs)
+        self.rest_handler.login()
+
+    def get_storage(self, context):
+        return self.rest_handler.get_storage(self.storage_id)
+
+    def list_storage_pools(self, context):
+        return self.rest_handler.get_storage_pools(self.storage_id)
+
+    def list_volumes(self, context):
+        return self.rest_handler.get_volumes(self.storage_id)
+
+    def list_alerts(self, context, query_para=None):
+        return self.rest_handler.list_alerts(query_para)
+
+    def clear_alert(self, context, alert):
+        pass
+
+    @staticmethod
+    def parse_alert(context, alert):
+        return rest_handler.RestHandler.get_parse_alerts(alert)
+
+    def get_alert_sources(self, context):
+        return self.rest_handler.get_alert_sources(self.storage_id)
+
+    def list_controllers(self, context):
+        return self.rest_handler.get_controllers(self.storage_id)
+
+    def list_disks(self, context):
+        return self.rest_handler.get_disks(self.storage_id)
+
+    def list_ports(self, context):
+        hardware_d = self.rest_handler.get_port_hardware()
+        fc_ports = self.rest_handler.get_fc_ports(self.storage_id, hardware_d)
+        fc_ports.extend(
+            self.rest_handler.get_eth_ports(self.storage_id, hardware_d))
+        fc_ports.extend(
+            self.rest_handler.get_sas_ports(self.storage_id, hardware_d))
+        return fc_ports
+
+    def reset_connection(self, context, **kwargs):
+        self.rest_handler.logout()
+        self.rest_handler.login()
+
+    def add_trap_config(self, context, trap_config):
+        pass
+
+    def remove_trap_config(self, context, trap_config):
+        pass
+
+    @staticmethod
+    def get_access_url():
+        return 'https://{ip}'
+
+    def collect_perf_metrics(self, context, storage_id, resource_metrics,
+                             start_time, end_time):
+        LOG.info('The system(storage_id: %s) starts to collect storage and'
+                 ' volume performance, start_time: %s, end_time: %s',
```

Comment:
The log in this place maybe like "The system(storage_id: %s) starts to collect performance, start_time: %s, end_time: %s".

Reply:
It has been modified.

LOG.info('The system(storage_id: %s) starts to collect performance,'
                 ' start_time: %s, end_time: %s',
                 storage_id, start_time, end_time)